urlpattern/resources/urlpatterntestdata.json

```diff
@@ -2419,15 +2436,24 @@
   },
   {
     "pattern": [{ "hostname": "bad\nhostname" }],
```

Comment:
Why are `\n`, `\r`, and `\t` just stripped?

Reply:
They are omitted because URL parser hostname state step 3 says

if url is special and c is U+005C (\) ... Let host be the result of host parsing buffer with url is not special.

Basically if you see `\` character you stop there and omit that character, and take the existing buffer and parse it as a valid hostname

Reply:
This is defined in URL spec. All ascii tab or newline are removed from the input before any parsing is done.

> Remove all [ASCII tab or newline](https://infra.spec.whatwg.org/#ascii-tab-or-newline) from input.